libsignal-protocol/src/crypto/mod.rs

```diff
@@ -398,3 +399,46 @@ unsafe extern "C" fn internal_cipher(
         Err(e) => e.code(),
     }
 }
+
+#[cfg(all(test, feature = "crypto-native", feature = "crypto-openssl"))]
+mod crypto_tests {
+    use super::*;
+
+    #[test]
+    fn test_crypter() {
+        // Here is a test to see the behavior of DefaultCrypto vs OpenSSLCrypto
+        let native_crypto = DefaultCrypto::default();
+        let openssl_crypto = OpenSSLCrypto::default();
+        let data = [1, 2, 3, 4, 5, 6, 7];
+        let mut key = [0u8; 16];
+        let mut iv = [0u8; 16];
+        native_crypto.fill_random(&mut key).unwrap();
+        native_crypto.fill_random(&mut iv).unwrap();
+
+        let cipher_text_native = native_crypto
+            .encrypt(SignalCipherType::AesCbcPkcs5, &key, &iv, &data)
+            .unwrap();
+        let cipher_text_openssl = openssl_crypto
+            .encrypt(SignalCipherType::AesCbcPkcs5, &key, &iv, &data)
+            .unwrap();
+        assert_eq!(cipher_text_native, cipher_text_openssl);
```

Comment:
Good idea making sure the native and OpenSSL implementations are identical!

Reply:
add both tests for `CBC/CTR` tests